aiida_cp2k/calculations/__init__.py

```diff
@@ -2,230 +2,160 @@
+        else:
+            settings = {}
 
         # create code info
         codeinfo = CodeInfo()
-        cmdline = settings.pop('cmdline', [])
-        cmdline += ["-i", self._INPUT_FILE_NAME]
-        codeinfo.cmdline_params = cmdline
-        codeinfo.stdout_name = self._OUTPUT_FILE_NAME
+        codeinfo.cmdline_params = settings.pop('cmdline', []) + ["-i", self._DEFAULT_INPUT_FILE]
+        codeinfo.stdout_name = self._DEFAULT_OUTPUT_FILE
         codeinfo.join_files = True
-        codeinfo.code_uuid = code.uuid
+        codeinfo.code_uuid = self.inputs.code.uuid
 
         # create calc info
         calcinfo = CalcInfo()
-        calcinfo.stdin_name = self._INPUT_FILE_NAME
+        calcinfo.stdin_name = self._DEFAULT_INPUT_FILE
         calcinfo.uuid = self.uuid
         calcinfo.cmdline_params = codeinfo.cmdline_params
-        calcinfo.stdin_name = self._INPUT_FILE_NAME
-        calcinfo.stdout_name = self._OUTPUT_FILE_NAME
+        calcinfo.stdin_name = self._DEFAULT_INPUT_FILE
+        calcinfo.stdout_name = self._DEFAULT_OUTPUT_FILE
         calcinfo.codes_info = [codeinfo]
 
         # file lists
         calcinfo.remote_symlink_list = []
-        calcinfo.local_copy_list = local_copy_list
+        if 'file' in self.inputs:
+            calcinfo.local_copy_list = []
+            for fobj in self.inputs.file.values():
+                calcinfo.local_copy_list.append((fobj.uuid, fobj.filename, fobj.filename))
+
         calcinfo.remote_copy_list = []
-        calcinfo.retrieve_list = [self._OUTPUT_FILE_NAME,
-                                  self._RESTART_FILE_NAME]
+        calcinfo.retrieve_list = [self._DEFAULT_OUTPUT_FILE, self._DEFAULT_RESTART_FILE_NAME]
         calcinfo.retrieve_list += settings.pop('additional_retrieve_list', [])
 
         # symlinks
-        if parent_calc_folder is not None:
-            comp_uuid = parent_calc_folder.get_computer().uuid
-            remote_path = parent_calc_folder.get_remote_path()
-            symlink = (comp_uuid, remote_path, self._PARENT_CALC_FOLDER_NAME)
+        if 'parent_calc_folder' in self.inputs:
+            comp_uuid = self.inputs.parent_calc_folder.computer.uuid
+            remote_path = self.inputs.parent_calc_folder.get_remote_path()
+            symlink = (comp_uuid, remote_path, self._DEFAULT_PARENT_CALC_FLDR_NAME)
             calcinfo.remote_symlink_list.append(symlink)
 
         # check for left over settings
         if settings:
-            msg = "The following keys have been found "
-            msg += "in the settings input node {}, ".format(self.pk)
-            msg += "but were not understood: " + ",".join(settings.keys())
-            raise InputValidationError(msg)
+            raise InputValidationError("The following keys have been found " +
+                                       "in the settings input node {}, ".format(self.pk) + "but were not understood: " +
+                                       ",".join(list(settings.keys())))
```

Comment:
`str.join()` works with iterators and views as well, no need to convert to a full list here.